### PR TITLE
Add ability to upload CA certificates to trust snapshot storage endpoints

### PR DESCRIPTION
**Pull request by alicenstar**
#### What type of PR is this?

type::feature

#### What this PR does / why we need it:

This PR adds the ability to upload CA certificates in the UI in order to trust snapshot storage providers.

- Adds the ability for the back end to accept a CA Certificate for upload
- Adds the ability to attach a CA Certificate to a provider when adding a snapshot destination

#### Which issue(s) this PR fixes:

Fixes https://app.shortcut.com/replicated/story/33336/kots-ui-ability-to-upload-ca-certificates-to-trust-snapshot-storage-endpoints

#### Does this PR introduce a user-facing change?

```release-note
Adds an "Add a CA Certificate" field to the snapshot page.
```
#### Does this PR require documentation?

Yes, new documentation

### CODE REVIEW

File: web/src/components/snapshots/ConfigureSnapshots.jsx [hunk 1]
```diff
@@ -2,8 +2,7 @@ import React from "react";
 import Modal from "react-modal";
```
Comment (alicenstar) [inline web/src/components/snapshots/ConfigureSnapshots.jsx:2]: All changes in this file are linting/formatting.

File: web/src/components/snapshots/SnapshotStorageDestination.jsx [hunk 2]
```diff
@@ -376,19 +348,25 @@ class SnapshotStorageDestination extends Component {
...
 
     if (!urlRe.test(this.state.s3CompatibleEndpoint)) {
-        return { "endpoint" : "Please enter a valid endpoint with protocol"}
+      return { "endpoint": "Please enter a valid endpoint with protocol" }
     }
     return {}
   }
 
   getProviderPayload = (provider, bucket, path) => {
+    const ca_cert_data = this.state.caCertificate.data;
```
[the remark this answers is not in the document]
Comment (alicenstar) [reply web/src/components/snapshots/SnapshotStorageDestination.jsx:357]: I was unsure, and based on looking at this line (https://github.com/replicatedhq/kots/pull/2683/files#diff-584931fb8aa3f357709bb709983f44b347fcee37bef8f9b8c0ea98f773acf378R69), I thought I had to name it that way if that is what the back end is expecting. That being said, I am unfamiliar with Go syntax, so I could be reading that wrong.

File: pkg/handlers/snapshots.go [hunk 3]
```diff
@@ -65,6 +65,8 @@ type UpdateGlobalSnapshotSettingsRequest struct {
 	Other      *kotssnapshottypes.StoreOther  `json:"other"`
 	Internal   bool                           `json:"internal"`
 	FileSystem *FileSystemOptions             `json:"fileSystem"`
+
+	CACertData []byte `json:"ca_cert_data"`
```
Comment (sgalsaleh) [inline pkg/handlers/snapshots.go:69]: let's keep the naming convenction
```suggestion
	CACertData []byte `json:"caCertData"`
```

File: web/src/components/snapshots/UploadCACertificate.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              </p>
+              <p className="u-fontSize--small u-textColor--info u-lineHeight--normal">
+                Supported file types are .pem, .cer, .crt, .ca, and .key
+              </p>
+            </div>
+          </div>
+        </Dropzone>
+      }
+
+      {/* TODO: Update this link to the docs!! */}
```
Comment (sgalsaleh) [inline web/src/components/snapshots/UploadCACertificate.jsx:77]: update this? :)

Comment (alicenstar) [reply web/src/components/snapshots/UploadCACertificate.jsx:77]: I thought maybe we should leave this in until the docs are actually ready to go. This piece can quickly be uncommented when those are ready, though if that's not typical protocol I am happy to remove the whole commented chunk :)

Comment (sgalsaleh) [reply web/src/components/snapshots/UploadCACertificate.jsx:77]: yeah, let's remove the comment and create a SC story to track this instead please.

**Review by sgalsaleh — APPROVED** [review]

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**